llvm/lib/Transforms/Vectorize/VPlan.cpp

```diff
@@ -1112,6 +1123,26 @@ VPlan *VPlan::duplicate() {
   return NewPlan;
 }
 
+void VPlan::setName(const VPValue *V, const Twine &Name) {
+#if !defined(NDEBUG) || defined(LLVM_ENABLE_DUMP)
+  std::string N = Name.str();
+  if (N.empty())
+    return;
+  assert(!VPValue2Name.contains(V));
+  std::string CurrName = N;
+
+  if (UsedNames.contains(N))
+    CurrName = N + ".1";
+  unsigned Cnt = 2;
+  while (UsedNames.contains(CurrName)) {
```

Comment:
This O(n^2) search is tolerable because n is expected to be very small? Can alternatively hold a map between used names and the number of times each name is used, instead of a set. The former will avoid reusing names that were removed.

Reply:
Yes, the number of clashes should be low.

> The former will avoid reusing names that were removed.
The code at the moment intentionally allows re-using names for values that have been removed.